src/log_prob.jl

```diff
@@ -0,0 +1,438 @@
+    fluxes = colors_to_fluxes(brightness, colors)
+
+    # make sure elbo-args reflects the position and galaxy shape passed in for
+    # the first source in the elbo args (first is current source, the rest are
+    # conditioned on)
+    ea.vp[1][ids.u[1]]    = position[1]
+    ea.vp[1][ids.u[2]]    = position[2]
+    ea.vp[1][ids.e_dev]   = gal_shape[1]
+    ea.vp[1][ids.e_axis]  = gal_shape[2]
+    ea.vp[1][ids.e_angle] = gal_shape[3]
+    ea.vp[1][ids.e_scale] = gal_shape[4]
+
+    # create objects needed to compute the mean poisson value per pixel
+    # (similar to ElboDeriv.process_active_pixels!)
+    elbo_vars =
+      ElboIntermediateVariables(Float64, ea.S, length(ea.active_sources))
+    elbo_vars.calculate_derivs = false
+    elbo_vars.calculate_hessian= false
+
+    # load star/gal mixture components
+    star_mcs_vec = Array(Array{BvnComponent{Float64}, 2}, ea.N)
+    gal_mcs_vec  = Array(Array{GalaxyCacheComponent{Float64}, 4}, ea.N)
+    for b=1:ea.N
+        star_mcs_vec[b], gal_mcs_vec[b] =
+            load_bvn_mixtures(ea, b,
+                              calculate_derivs=elbo_vars.calculate_derivs,
+                              calculate_hessian=elbo_vars.calculate_hessian)
+    end
+
+    # iterate over the pixels, summing pixel-specific poisson rates
+    ll = 0.
+    for pixel in active_pixels
+        tile         = ea.images[pixel.n].tiles[pixel.tile_ind]
+        tile_sources = ea.tile_source_map[pixel.n][pixel.tile_ind]
+        this_pixel   = tile.pixels[pixel.h, pixel.w]
+        pixel_band   = tile.b
+
+        # compute the unit-flux pixel values
+        populate_fsm_vecs!(elbo_vars, ea, tile_sources, tile,
+                           pixel.h, pixel.w,
+                           gal_mcs_vec[pixel.n], star_mcs_vec[pixel.n])
+
+        # compute the background rate for this pixel
+        background_rate = tile.epsilon_mat[pixel.h, pixel.w]
+        background_sources = tile_sources[tile_sources.!=1]
+        for s in background_sources
+            println("background s: ", s)
+            flux_s = variational_params_to_fluxes(s, ea.vp)
+            rate_s = is_star ? elbo_vars.fs0m_vec[s].v : elbo_vars.fs1m_vec[s].v
+            rate_s *= flux_s[pixel_band]
+            background_rate += rate_s
+        end
+
+        # this source's rate, add to background for total
+        this_rate  = is_star ? elbo_vars.fs0m_vec[1].v : elbo_vars.fs1m_vec[1].v
+        pixel_rate = fluxes[pixel_band]*this_rate + background_rate
+
+        # multiply by image's gain for this pixel
+        rate     = pixel_rate * tile.iota_vec[pixel.h]
+        pixel_ll = logpdf(Poisson(rate[1]), round(Int, this_pixel))
+        ll += pixel_ll
+    end
+    return ll
+end
+
+
+###########################################################################
+# priors - for computing prior log probs of star/gal parameters           #
+###########################################################################
+
+
+type StarPrior
```

Comment:
Better organization would be to call these ```MCMCStarPrior``` &c and put them in a ```priors.jl``` file with the other prior types, which are now in ```light_source_model.jl```, probably by historical accident.